embeddings/data/datamodule.py

```diff
@@ -0,0 +1,167 @@
+import abc
+from typing import Any, Dict, Generic, List, Optional, TypeVar, Union
+
+import datasets
+import pytorch_lightning as pl
+from datasets import DatasetDict
+from torch.utils.data import DataLoader
+from transformers import AutoTokenizer, BatchEncoding
+
+from embeddings.utils.loggers import get_logger
+
+Data = TypeVar("Data")
+HuggingFaceDataset = TypeVar("HuggingFaceDataset")
+
+_logger = get_logger(__name__)
+
+
+class BaseDataModule(abc.ABC, pl.LightningDataModule, Generic[Data]):
+    dataset: Data
+
+
+class HuggingFaceDataModule(BaseDataModule[DatasetDict]):
+    loader_columns = [
+        "datasets_idx",
+        "input_ids",
+        "token_type_ids",
+        "attention_mask",
+        "start_positions",
+        "end_positions",
+        "labels",
+    ]
+    num_labels: int
+
+    def __init__(
+        self,
+        model_name_or_path: str,
+        dataset_name: str,
+        target_field: str,
+        max_seq_length: int = 128,
```

Comment:
Is it possible to set `max_seq_length` and `input_dim` automatically from model? This may cause unnecessary bugs

Reply:
max_seq_length is used in tokenizer to cut off lengthy sentence; I'll remove the input_dim since it was only used for the custom model to set the head of classifier

Reply:
But this is limited by the model, and if I rembember that correctly for bert is `512`. Other models may handle longer sequences. So for my natural chocie would be `max_sequence_length` supported by model or passing custom value

Reply:
ok you're right, i'll set it to None for a default so it'll take the maximum length supported by the model